libs/broker-db/src/converter.ts

```diff
@@ -0,0 +1,50 @@
+import { DocumentData, FirestoreDataConverter, toDateTime, toTimestamp } from "@thecointech/firestore";
+import { Decimal } from "decimal.js-light";
+import { isPlainObject } from 'lodash';
+
+function convertObject(obj: DocumentData, keys: string[], converter: (v: any) => any) {
+  // do a deep clone
+  const r = { ...obj };
+  Object.keys(obj).forEach(k => {
+    if (keys.includes(k)) {
+      r[k] = converter(obj[k]);
+    }
+    else if (isPlainObject(obj[k])) {
+      r[k] = convertObject(obj[k], keys, converter);
+    }
+  })
+  return r;
+}
+type converter = {
+  toFirestore: (v: any) => any;
+  fromFirestore: (v: any) => any;
```

Comment:
I don't like any's but I couldn't figure out how to do these as a template, and they are not export so the `any` isn't likely to cause confusion